sphinx_doc/howtos/tutorials/docs/navigation2_with_stvl.rst

```diff
@@ -0,0 +1,148 @@
+.. _navigation2-on-real-turtlebot3:
+
+Navigation2 With External Costmap Plugin
+****************************************
+
+- `Overview`_
+- `Costmap2D and STVL`_
+- `Tutorial Steps`_
+- `Videos`_
+
+Overview
+========
+
+This tutorial shows how to load and use an external plugin.
+This example uses the `Spatio Temporal Voxel Layer <https://github.com/SteveMacenski/spatio_temporal_voxel_layer/>`_ (STVL) costmap pluginlib plugin as an example.
+STVL is a demonstrative pluginlib plugin and the same process can be followed for other costmap plugins as well as plugin planners, controllers, and recoveries.
+
+Before completing this tutorial, please look at the previous two tutorials on navigation in simulation and physical hardware, if available.
+This tutorial assumes knowledge of navigation and basic understanding of costmaps.
+
+Costmap2D and STVL
+==================
+
+Costmap 2D is the software we use to buffer sensor information into a global view that the robot will use to create plans and control efforts.
+Within Costmap2D, there are pluginlib plugin interfaces available to create custom behaviors loadable at runtime.
```

Comment:
maybe a link to pluginlib documentation here?

Also 'pluginlib plugin' seems redundantly redundant.

Reply:
Should reading the pluginlib wiki be a pre-requisite?

Reply:
Sure, I can add the link.

Given this is discussing how to use an existing plugin and not how to create one, I think it would be a little off topic to require the reading, but a link is good. 

I say pluginlib plugin to make sure people know about this thing “pluginlib” which isn’t always something people run into until the navigation stack or moveit.

Reply:
OK, I agree it doesn't need to be a pre-requisite, that makes sense. I think a link will also help clarify the redundancy. Thanks!

Reply:
Added